src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -59,7 +66,63 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
         {
             lock (_gate)
             {
-                _openDocumentContexts.Add(e.Document.Id, Task.Run(() => GetConventionContextAsync(e.Document.FilePath, CancellationToken.None)));
+                var documentId = e.Document.Id;
+                var filePath = e.Document.FilePath;
+                _openDocumentContexts.Add(documentId, Task.Run(async () =>
+                {
+                    var context = await GetConventionContextAsync(filePath, CancellationToken.None).ConfigureAwait(false);
+                    context.CodingConventionsChangedAsync += (innerSender, innerE) => HandleCodingConventionsChangedAsync(documentId, innerSender, innerE);
```

Comment:
so, now, not only we leak context, we leak event handler? how this get unsubscribed?

Reply:
There is no explicit unsubscribe. However:

1. This is an instance event, not a static event. The handler should not extend past the life of the owner, which is released in `Workspace_DocumentClosed`¹


¹ You mentioned there may be a bug where this is not always the case, but that bug is not _introduced_ by this pull request and will be corrected independently.

Reply:
I am not sure why not fixing leaks when you change code this much...